### DISCUSSION TITLE
Postgres Maximum Schemas and Tables

### DISCUSSION THREAD

**Discussion by LM3Any** [supabase/supabase#17212] [Questions]
Hello, In the pro plan other than the 8GB database size limitation are there any hard limits on the following?

- Database maximum schemas
- Maximum tables within one schema

Thank you for your help.

**Comment by GaryAustin1** [comment]
No hard limits on either.  Just space.
Edit: And complexity of grants/RLS on lots of entities if needed.

**Reply by LM3Any** [reply]
Thank you. I am weighing how to support multi-tenant. There could be 10s of 1000s.

I have thought of using a different schema for each tenant but not sure yet. Each schema would have only a few tables. A problem might be references - not FK - across schemas for some document sharing between tenants - like google drive and docs. All tables in every schema would be identical so admin would be reduced. Another option is creating unique tables for each tenant in the same schema.

I have read that many schemas can be difficult to back up and restore and that schema resource caching can cause memory problems. Others are not concerned about that. Who is more correct?

I know that RLS can be used for all tenants using the same tables - but I read that RLS comes with a lot of admin risk. I have used the tenant id as a discriminator in tables but I don't like the risk of wrong queries and I want to assure users that their data is separate from others. If there is a lot of admin work for RLS I do not want that for many tenants.

Opinions are OK but prefer real experience. Thank you again for your help.
